hudi-spark/src/main/scala/org/apache/hudi/IncrementalRelation.scala

```diff
@@ -71,13 +78,16 @@ class IncrementalRelation(val sqlContext: SQLContext,
     optParams.getOrElse(DataSourceReadOptions.END_INSTANTTIME_OPT_KEY, lastInstant.getTimestamp))
     .getInstants.iterator().toList
 
-  // use schema from latest metadata, if not present, read schema from the data file
-  private val latestSchema = {
-    val schemaUtil = new TableSchemaResolver(metaClient)
-    val tableSchema = HoodieAvroUtils.createHoodieWriteSchema(schemaUtil.getTableAvroSchemaWithoutMetadataFields);
-    AvroConversionUtils.convertAvroSchemaToStructType(tableSchema)
+  // use schema from a file produced in the latest instant
+  val latestSchema: StructType = {
+    log.info("Inferring schema..")
+    val schemaResolver = new TableSchemaResolver(metaClient)
+    val tableSchema = schemaResolver.getTableAvroSchemaWithoutMetadataFields
+    val dataSchema = AvroConversionUtils.convertAvroSchemaToStructType(tableSchema)
+    StructType(skeletonSchema.fields ++ dataSchema.fields)
```

Comment:
Not really. The reason for doing it this way is to intentionally only read the **user data schema** and then append the **metadata/skeleton schema** to it. This avoids us to have unnecessary checks here, because if we read the whole schema then there will be differences. For regular hudi files, the schema would have both **skeleton + user data schema** whereas for **bootstrapped files** the schema would only have **user data schema** read from the source file. So to keep things simple I did it this way.